api/validator.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+__author__ = 'Øyvind Jekteberg and Kristian Gingstad'
+__copyright__ = 'Copyright 2018, The ArXivDigest Project'
+
+from functools import wraps
+from flask import Flask, g, jsonify, request, make_response
+from flask import current_app as app
+import api.database as db
+from datetime import datetime
+
+
+def validate_json(validator_func):
+    '''Decorator for validating submitted json using supplied validator function.
+       Validator functions should take a json as input argument, and return none if valid,
+       or a response if something is invalid.'''
+    def decorator(f):
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            json = request.get_json()
+            if not json:
+                return make_response(jsonify({'success': False, 'error': 'No JSON submitted.'}), 400)
+            error = validator_func(json)
+            if error:
+                return error
+            return f(*args, **kwargs)
+        return wrapper
+    return decorator
+
+
+def recommendation(json):
```

Comment:
This method is getting too long. It would be nice to refactor each of the checks into separate methods.